cmd/experimental/kjobctl/pkg/cmd/describe/describe_printer.go

```diff
@@ -0,0 +1,127 @@
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package describe
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/duration"
+)
+
+// ResourceDescriber generates output for the named resource or an error
+// if the output could not be generated. Implementers typically
+// abstract the retrieval of the named object from a remote server.
+type ResourceDescriber interface {
+	Describe(object *unstructured.Unstructured) (output string, err error)
+}
+
+// Describer returns a Describer for displaying the specified RESTMapping type or an error.
+func NewResourceDescriber(mapping *meta.RESTMapping) (ResourceDescriber, error) {
+	if describer, ok := DescriberFor(mapping.GroupVersionKind.GroupKind()); ok {
+		return describer, nil
+	}
+
+	return nil, fmt.Errorf("no description has been implemented for %s", mapping.GroupVersionKind.String())
+}
+
+// DescriberFor returns the default describe functions for each of the standard
+// Kubernetes types.
+func DescriberFor(kind schema.GroupKind) (ResourceDescriber, bool) {
+	describers := map[schema.GroupKind]ResourceDescriber{
+		{Group: batchv1.GroupName, Kind: "Job"}: &JobDescriber{},
+	}
+
+	f, ok := describers[kind]
+	return f, ok
+}
+
+// JobDescriber generates information about a job and the pods it has created.
+type JobDescriber struct{}
+
+func (d *JobDescriber) Describe(object *unstructured.Unstructured) (string, error) {
+	job := &batchv1.Job{}
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(object.UnstructuredContent(), job)
+	if err != nil {
+		return "", err
+	}
+
+	return describeJob(job)
+}
+
+func describeJob(job *batchv1.Job) (string, error) {
+	return tabbedString(func(out io.Writer) error {
+		w := NewPrefixWriter(out)
+		w.Write(IndentLevelZero, "Name:\t%s\n", job.Name)
+		w.Write(IndentLevelZero, "Namespace:\t%s\n", job.Namespace)
+		if selector, err := metav1.LabelSelectorAsSelector(job.Spec.Selector); err == nil {
+			w.Write(IndentLevelZero, "Selector:\t%s\n", selector)
+		} else {
+			w.Write(IndentLevelZero, "Selector:\tFailed to get selector: %s\n", err)
+		}
+		printLabelsMultiline(w, "Labels", job.Labels)
+		printAnnotationsMultiline(w, "Annotations", job.Annotations)
+		if controlledBy := printController(job); len(controlledBy) > 0 {
+			w.Write(IndentLevelZero, "Controlled By:\t%s\n", controlledBy)
+		}
```

Comment:
Will it make sense to only print the values coming for the profile's  `RequiredFlags`

Reply:
Not sure that we should print only RequiredFlags as we're going to have 5 properties max. Maybe we can make sure that those in the required flags are printed too but it might be useful to show the other properties as well.

Reply:
List of properties printed has been updated. 

CC: @mwielgus